matrix/Matrix.hpp

```diff
@@ -35,12 +35,12 @@ class Matrix
     // Constructors
     Matrix() = default;
 
-    Matrix(const Type data_[][N])
+    Matrix(const Type data_[M])
```

Comment:
TODO review
 - Mx1 constructor (vector)
 - do we need to support MxN matrix construction from an array that's M*N elements